includes/Core/Modules/REST_Modules_Controller.php

```diff
@@ -0,0 +1,645 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Modules\REST_Modules_Controller
+ *
+ * @package   Google\Site_Kit\Core\Modules
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Modules;
+
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\REST_API\REST_Routes;
+use Google\Site_Kit\Core\REST_API\REST_Route;
+use Google\Site_Kit\Core\REST_API\Exception\Invalid_Datapoint_Exception;
+use WP_REST_Server;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Error;
+use Exception;
+
+/**
+ * Class for handling modules rest routes.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class REST_Modules_Controller {
+
+	const REST_ENDPOINT_CHECK_ACCESS = 'core/modules/data/check-access';
+
+	/**
+	 * Modules instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Modules
+	 */
+	protected $modules;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Modules $modules Modules instance.
+	 */
+	public function __construct( Modules $modules ) {
+		$this->modules = $modules;
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_filter(
+			'googlesitekit_rest_routes',
+			function( $routes ) {
+				return array_merge( $routes, $this->get_rest_routes() );
+			}
+		);
+
+		add_filter(
+			'googlesitekit_apifetch_preload_paths',
+			function ( $paths ) {
+				$active_modules = $this->modules->get_active_modules();
```

Comment:
This line is unused because it's called again below 🙂 
```suggestion
```

Reply:
👍